migration/idempotent/001-base.sql

```diff
@@ -3316,3 +3316,35 @@ END;
 $$
 LANGUAGE PLPGSQL;
 GRANT EXECUTE ON FUNCTION _prom_catalog.insert_metric_row(TEXT, TIMESTAMPTZ[], DOUBLE PRECISION[], BIGINT[]) TO prom_writer;
+
+-- Creates a temporary table (if it doesn't exist) used for ingestion of metrics and traces
+-- Supresses corresponding DDL logging, otherwise PG log may get unnecessarily verbose.
+-- Temporary table is created using supplied table and schema as prototype.
+-- Returns temporary table name
+CREATE OR REPLACE FUNCTION _prom_catalog.create_ingest_temp_table(table_name TEXT, schema_name TEXT)
+    RETURNS TEXT
+    SECURITY DEFINER
+    VOLATILE
+    SET search_path = pg_catalog, pg_temp
+AS $func$
+DECLARE
+    temp_table TEXT;               
+BEGIN
+    SET LOCAL log_statement = 'none';
+    -- since table_name gets sanitized by jackc/pgx we need to remove quotes
+    temp_table := FORMAT('%s_temp',REPLACE(table_name, '"', ''));
```

Comment:
I took a deeper look at how this function behaves, and I suspect that by appending `_temp` to the table name, we violate some assumptions about how we handle long metric names (longer than 62 chars).

I think that we should not append `_temp` to the temporary table name. It is uniquely addressable by using the `pg_temp` schema.

Reply:
Yes, that's what @jgpruitt already recommended.

Reply:
Hmmm... Okay I must have missed that...